src/replica.rs

```diff
@@ -210,11 +215,18 @@ impl Replica {
     }
 
     /// Step 2. Validation of agreement, and order at debit source.
-    pub fn register(&self, debit_proof: &DebitAgreementProof) -> Result<TransferRegistered> {
+    pub fn register<F: FnOnce() -> bool>(
+        &self,
+        debit_proof: &DebitAgreementProof,
+        f: F,
```

Comment:
`f` is a very obscure naming. I had no idea what this was until I drilled down to where it is executed, and there was a comment.